docs/samples/Microsoft.ML.Samples/Dynamic/IidSpikeDetectorTransform.cs

```diff
@@ -83,5 +87,84 @@ public static void IidSpikeDetectorTransform()
             // 0       5.00    0.50
             // 0       5.00    0.50
         }
+
+        public static void IidSpikeDetectorPrediction()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Generate sample series data with a spike
+            const int size = 10;
+            var data = new List<IidSpikeData>(size);
+            for (int i = 0; i < size / 2; i++)
+                data.Add(new IidSpikeData(5));
+            // This is a spike
+            data.Add(new IidSpikeData(10));
+            for (int i = 0; i < size / 2; i++)
+                data.Add(new IidSpikeData(5));
+
+            // Convert data to IDataView.
+            var dataView = ml.CreateStreamingDataView(data);
+
+            // Setup IidSpikeDetector arguments
+            string outputColumnName = "Prediction";
+            string inputColumnName = "Value";
+            var args = new IidSpikeDetector.Arguments()
+            {
+                Source = inputColumnName,
+                Name = outputColumnName,
+                Confidence = 95,                // The confidence for spike detection in the range [0, 100]
+                PvalueHistoryLength = size / 4  // The size of the sliding window for computing the p-value
```

Comment:
>size / 4  [](start = 38, length = 9)

Any reason why we divide by 4?
Is it magic number, or it has some useful meaning?

Reply:
You can have any length of history. I'm just replicating the above example here to show it can also be used with prediction engine.

---
In reply to: [237330349](https://github.com/dotnet/machinelearning/pull/1762#discussion_r237330349) [](ancestors = 237330349)

Reply:
Does it help to have a comment something like this: "The size of the sliding window for computing the p-value; shorter windows are more sensitive to spikes."

---
In reply to: [237623190](https://github.com/dotnet/machinelearning/pull/1762#discussion_r237623190) [](ancestors = 237623190,237330349)